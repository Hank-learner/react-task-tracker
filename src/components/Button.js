import PropTypes from 'prop-types'

const Button = ({ color, text, onClick, dataTestID }) => {
    return (
        <button
            onClick={onClick}
            className='btn'
            style={{ backgroundColor: color }}
            data-testid={dataTestID}
        >
            {text}
        </button>
    )
}

Button.defaultProps = {
    color: 'steelblue',
    dataTestID: 'testing',
    text:'NULL',
    onClick: () => { console.log('button not assigned onClick') }
}

Button.propTypes = {
    text: PropTypes.string,
    color: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    dataTestID: PropTypes.string
}

export default Button
