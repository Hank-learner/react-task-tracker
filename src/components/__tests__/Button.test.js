import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from '../Button';

const mockOnAdd = jest.fn()
jest.mock('../../App', () => {
    return jest.fn(() => ({
        onAdd: mockOnAdd
    }))
})

const validProps = {
    color:'brown',
    text : 'Add',
    onClick: mockOnAdd,
    dataTestID:'add-task'
}

describe('Button snapshot test', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<Button {...validProps} text/>)
        expect(asFragment(<Button {...validProps} />)).toMatchSnapshot()
    });
});

test('can click add btn', () => {
    const { getByTestId } = render(<Button {...validProps}/>)
    fireEvent.click(getByTestId('add-task'))
    expect(mockOnAdd).toBeCalled()
})

// describe('render testing', () => {

//     test('renders addTask', () => {
//         const { getByText } = renderer.create(Button).toJSON();
//         expect(getByText('Developed with help of Traversy Media Â© 2021', { exact: false })).not.toBeNull();
//     });
// })