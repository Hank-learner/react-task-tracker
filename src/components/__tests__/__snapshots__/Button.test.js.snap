// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot test renders correctly 1`] = `
<DocumentFragment>
  <button
    class="btn"
    data-testid="add-task"
    style="background-color: brown;"
  />
</DocumentFragment>
`;
